vowpalwabbit/debug_log.h

```diff
@@ -2,9 +2,50 @@
 // individual contributors. All rights reserved. Released under a BSD (revised)
 // license as described in the file LICENSE.
 
+#include "future_compat.h"
 #pragma once
 
-#define VW_DEBUG_ENABLE(x) static bool VW_DEBUG_LOG = x;
+struct vw_dbg
+{
+  static constexpr bool default_log         = false;
+  static constexpr bool learner             = false;
+  static constexpr bool search              = false;
+  static constexpr bool gd                  = false;
+  static constexpr bool gd_predict          = false;
+  static constexpr bool binary              = false;
+  static constexpr bool cb_adf              = false;
+  static constexpr bool csoaa               = false;
+  static constexpr bool cs_active           = false;
+  static constexpr bool cats_tree           = false;
+  static constexpr bool cats_pdf            = false;
+  static constexpr bool cats                = false;
+  static constexpr bool cb_explore_pdf      = false;
+  static constexpr bool cb_explore          = false;
+  static constexpr bool cbify               = false;
+  static constexpr bool get_pmf             = false;
+  static constexpr bool sample_pdf          = false;
```

Comment:
so you have to manually add a boolean here that sort of matches the name used in that file that sort of matches the string name passed to learner? isn't that a bit excessive

Reply:
These booleans define different scopes that can be enabled or not.  If you don't need a new scope it uses the default_log scope.